manifests/simple/rule.pp

```diff
@@ -14,14 +14,13 @@
 # begin currently ignored
   Array $urls = [],
 # end currently ignored
-  Array $examples = [],
+  Array[Patterndb::Example] $examples = [],
   Array $tags = [],
   Hash $values = {},
 ) {
   $patterns_sanitized = htmlentities($patterns)
 
   # validate sample messages
-  patterndb_simple_example ($examples, $id)
```

Comment:
The class `patterndb::simple::example` did not create resources, it only did validation, so was removed.
The `patterndb_simple_example()` function was just creating these resources so it was also removed.
Se the call to the function was also removed.

Now the following manifest will fail instead of doing nothing:
```puppet
patterndb::simple::example { 'foo':
  # ...
}
```

The correct way to use this was to pass parameters to the `patterndb::simple::rule`, and now the validation is done there, so we do not loose functionality.


I see we have the same for `patterndb::simple::action::message` which I missed, will add a commit to remove it too!

Reply:
> I see we have the same for `patterndb::simple::action::message` which I missed, will add a commit to remove it too!

Done in f6314a573cdbc3461c811ad0a518fdc7a99f4c24